tree/tree/src/TChain.cxx

```diff
@@ -402,7 +402,19 @@ Int_t TChain::Add(const char* name, Long64_t nentries /* = TTree::kMaxEntries */
       TObjString *obj;
       while ((obj = (TObjString*)next())) {
          file = obj->GetName();
-         nf += AddFile(TString::Format("%s/%s%s",directory.Data(),file,suffix.Data()),nentries);
+         if (suffix == "#" + treename) {
```

Comment:
Should we cache the result of `"#" + treename` instead of reacreating (with a potential memory allocation) at each iteration?

Reply:
Yes, that's a good idea thanks!